### PR TITLE
remove version check for grpc.

### PR DESCRIPTION
**Pull request by morvencao**
ref: https://issues.redhat.com/browse/ACM-10959

### CODE REVIEW

**Comment by morvencao** [comment]
test results:
```bash
# make test-integration
OCM_ENV=testing gotestsum --jsonfile-timing-events=/root/workspace/openshift-online-maestro/integration-test-results.json --format short-verbose -- -p 1 -ldflags -s -v -timeout 1h  \
		./test/integration
I0410 05:33:09.162145  159709 framework.go:76] Initializing testing environment
I0410 05:33:09.405651  159709 framework.go:161] Using Mock OCM Authz Client
I0410 05:33:09.407656  159709 framework.go:199] Disabling Sentry error reporting
I0410 05:33:09.417631  159709 grpc_server.go:70] Serving gRPC service without TLS at 8090
I0410 05:33:09.417909  159709 api_server.go:151] Serving without TLS at 8000
PASS test/integration.TestConsumerGet (0.11s)
PASS test/integration.TestConsumerPost (0.08s)
PASS test/integration.TestConsumerPatch (0.08s)
PASS test/integration.TestConsumerPaging (0.09s)
PASS test/integration.TestControllerRacing (1.61s)
PASS test/integration.TestControllerReconcile (1.04s)
PASS test/integration.TestControllerSync (1.06s)
PASS test/integration.TestPulseServer (6.06s)
PASS test/integration.TestResourceGet (0.05s)
PASS test/integration.TestResourcePost (2.05s)
PASS test/integration.TestResourcePatch (0.10s)
PASS test/integration.TestResourcePaging (0.27s)
PASS test/integration.TestResourceListSearch (0.15s)
PASS test/integration.TestUpdateResourceWithRacingRequests (0.15s)
PASS test/integration.TestResourceFromGRPC (3.12s)
PASS test/integration.TestResourceBundleFromGRPC (3.07s)
I0410 05:33:28.490556  159709 api_server.go:157] Web server terminated
I0410 05:33:28.491000  159709 healthcheck_server.go:60] HealthCheck server terminated
E0410 05:33:28.490824  159709 baseclient.go:90] the cloudevents client is disconnected, read tcp 127.0.0.1:58428->127.0.0.1:1883: use of closed network connection
E0410 05:33:28.491787  159709 logger.go:129]   failed to acquire the lock as another maestro instance is checking instances
PASS test/integration

DONE 16 tests in 22.614s
```

File: cmd/maestro/server/grpc_server.go [hunk 1]
```diff
@@ -133,11 +133,10 @@ func (svr *GRPCServer) Publish(ctx context.Context, pubReq *pbv1.PublishRequest)
...
 			}
-			// handle the special case that the resource is updated by the source controller
-			// and the version of the resource in the request is less than it in the database
-			if found.Version < res.Version {
-				res.Version = found.Version
-			}
+			// keep the existing version for bundle resource, mainly from hub controller,
+			// the version is not guaranteed to be increased.
+			// TODO: handle the resource version for bundle resource from the source
+			res.Version = found.Version
```
Comment (skeeey) [inline cmd/maestro/server/grpc_server.go:139]: we may need compare the resource manifests instead of comparing version?

Comment (morvencao) [reply cmd/maestro/server/grpc_server.go:139]: update method of resource service will check the manifest equality, do we need to do it here?
https://github.com/openshift-online/maestro/blob/1e08a6a7cb942379ca04a49be4fe5a8616e122d6/pkg/services/resource.go#L102-L104

**Pull request merged**